sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_content_understanding_content_analyzers_operations_async.py

```diff
@@ -0,0 +1,762 @@
+# pylint: disable=line-too-long,useless-suppression
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) Python Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+import pytest
+import os
+import re
+from typing import Tuple, Union, Dict, Any, Optional, cast
+from devtools_testutils.aio import recorded_by_proxy_async
+from testpreparer_async import ContentUnderstandingClientTestBaseAsync, ContentUnderstandingPreparer
+from azure.ai.contentunderstanding.models import ContentAnalyzer
+from azure.ai.contentunderstanding.aio import ContentUnderstandingClient
+from test_helpers import (
+    generate_analyzer_id_async,
+    extract_operation_id_from_poller,
+    new_simple_content_analyzer_object,
+    new_marketing_video_analyzer_object,
+    assert_poller_properties,
+    assert_simple_content_analyzer_result,
+    save_analysis_result_to_file,
+    save_keyframe_image_to_file,
+    PollerType,
+)
+from devtools_testutils import is_live, is_live_and_not_recording
+
+
+async def analyzer_in_list_async(client, analyzer_id: str) -> bool:
+    """Check if an analyzer with the given ID exists in the list of analyzers (async version).
+
+    Args:
+        client: The ContentUnderstandingClient instance
+        analyzer_id: The analyzer ID to search for
+
+    Returns:
+        bool: True if the analyzer is found, False otherwise
+    """
+    response = client.content_analyzers.list()
+    async for r in response:
+        if hasattr(r, "analyzer_id") and r.analyzer_id == analyzer_id:
+            return True
+    return False
+
+
+async def create_analyzer_and_assert_async(
+    client, analyzer_id: str, resource: Union[ContentAnalyzer, Dict[str, Any]]
+) -> Tuple[Any, str]:
+    """Create an analyzer and perform basic assertions (async version).
+
+    Args:
+        client: The ContentUnderstandingClient instance
+        analyzer_id: The analyzer ID to create
+        resource: The analyzer resource (ContentAnalyzer object or dict)
+
+    Returns:
+        Tuple[Any, str]: A tuple containing (poller, operation_id)
+
+    Raises:
+        AssertionError: If the creation fails or assertions fail
+    """
+    print(f"\nCreating analyzer {analyzer_id}")
+
+    # Start the analyzer creation operation
+    poller = await client.content_analyzers.begin_create_or_replace(
+        analyzer_id=analyzer_id,
+        resource=resource,
+    )
+
+    # Extract operation_id from the poller using the helper function
+    operation_id = extract_operation_id_from_poller(poller, PollerType.ANALYZER_CREATION)
```

Comment:
We should be using poller methods to check status instead of get_operation_status().